numpy/compat/py3k.py

```diff
@@ -7,15 +7,12 @@
 __all__ = ['bytes', 'asbytes', 'isfileobj', 'getexception', 'strchar',
            'unicode', 'asunicode', 'asbytes_nested', 'asunicode_nested',
            'asstr', 'open_latin1', 'long', 'basestring', 'sixu',
-           'integer_types', 'is_pathlib_path', 'npy_load_module', 'Path',
+           'integer_types', 'is_pathlib_path', 'npy_load_module',
            'pickle', 'contextlib_nullcontext', 'os_fspath', 'os_PathLike']
 
 import sys
 import os
-try:
-    from pathlib import Path, PurePath
-except ImportError:
-    Path = PurePath = None
+from warnings import warn
```

Comment:
This change breaks code that does
```
from numpy.compat import Path
if Path is None:
   ...
```

Do we care?

Reply:
yes we probably care. :(

Reply:
Maybe the only way to cleanup the namespace is to wait until python 3.7 is the only one you support, and then go all out with the deprecations using `__getattr__`. There is little sense giving warnings to users of python 3.7 + 3.8 only.

Reply:
> sense giving warnings to users of python 3.7 + 3.8 only.

Sure there is, it means that packages using CI across multiple versions can fix their code sooner :)

Reply:
Anyway, let's leave the slow import alone for now - we can still deprecate the function below.

Reply:
Although I'm starting to wonder whether removing useless functions from `compat` is ever a good idea, as it limits forward compatibility of package that still support python 2.

Reply:
yeah, i feel you. These kinds of warnings do get annoying.

Reply:
If nothing else, a change to this file to remove the `try` / `except` now that `pathlib` is always available would be appreciated

Reply:
There are also ALOT of places that have if/else statements like `sys.version > (3, 6)`, want to remove all of those if else clauses?

Reply:
lets leave it at pathlib for this PR.

Reply:
If you remove all the `> (3, 5)`  (preferably just from this file), then we can merge this in time for 1.18

Reply:
whoa, didn't know you all were still going to support 3.5 ;) Not many changes in this case for now. Many if staements can be removed. I'll leave it till then.